projects/plugins/jetpack/modules/carousel/jetpack-carousel.php

```diff
@@ -380,7 +380,7 @@ public function filter_gallery_block_render( $block_content, $block ) {
 
 		$this->enqueue_assets();
 
-		if ( ! isset( $post ) ) {
+		if ( ! isset( $post ) || ! $post instanceof WP_Post ) {
```

Comment:
We can simplify this
```suggestion
		if ( ! $post instanceof WP_Post ) {
```
`$post` will always be defined, even if it wasn't earlier, thanks to the `global $foo` above. `instanceof` will handle any value of `$post` correctly.